tests/Fixer/PhpUnit/PhpUnitTestAnnotationFixerTest.php

```diff
@@ -0,0 +1,427 @@
+class Test extends \PhpUnit\FrameWork\TestCase
+{
+    public function test_it_has_snake_case() {}
+}',
+                ['style' => 'annotation'],
+            ],
+            'Annotation is removed, and it is snake case' => [
+                '<?php
+class Test extends \PhpUnit\FrameWork\TestCase
+{
+    /**
+     */
+    public function test_it_has_snake_case() {}
+
+    public function helper_function() {}
+}',
+                '<?php
+class Test extends \PhpUnit\FrameWork\TestCase
+{
+    /**
+     * @test
+     */
+    public function it_has_snake_case() {}
+
+    public function helper_function() {}
+}',
+                ['case' => 'snake'],
+                ],
+                'Annotation gets added, it has an @depends, and we use snake case' => [
+                    '<?php
+class Test extends \PhpUnit\FrameWork\TestCase
+{
+    /**
+     * @test
+     */
+    public function works_fine () {}
+
+    /**
+     * @test
+     * @depends works_fine
+     */
+    public function works_fine_too() {}
+}',
+                    '<?php
+class Test extends \PhpUnit\FrameWork\TestCase
+{
+    public function test_works_fine () {}
+
+    /**
+     * @depends test_works_fine
+     */
+    public function test_works_fine_too() {}
+}',
+                    ['style' => 'annotation'],
+                    ],
+                'Annotation gets removed, it has an @depends and we use camel case' => [
+                    '<?php
+class Test extends \PhpUnit\FrameWork\TestCase
+{
+    /**
+     */
+    public function test_works_fine () {}
+
+    /**
+     * @depends test_works_fine
+     */
+    public function test_works_fine_too() {}
+}',
+                        '<?php
+class Test extends \PhpUnit\FrameWork\TestCase
+{
+    /**
+     * @test
+     */
+    public function works_fine () {}
+
+    /**
+     * @depends works_fine
+     */
+    public function works_fine_too() {}
+}',
+                    ['case' => 'snake'],
+                ],
+            'Class has both camel and snake case, annotated functions and not, and wants to add annotations' => [
+                '<?php
+class Test extends \PhpUnit\FrameWork\TestCase
+{
+    /**
+     * @test
+     */
+    public function snake_cased () {}
+
+    /**
+     * @test
+     */
+    public function camelCased () {}
+
+    /**
+     * @test
+     * @depends camelCased
+     */
+    public function depends_on_someone () {}
+
+    //It even has a comment
+    public function a_helper_function () {}
+
+    /**
+     * @test
+     * @depends depends_on_someone
+     */
+    public function moreDepends() {}
+}',
+                '<?php
+class Test extends \PhpUnit\FrameWork\TestCase
+{
+    public function test_snake_cased () {}
+
+    public function testCamelCased () {}
+
+    /**
+     * @depends testCamelCased
+     */
+    public function test_depends_on_someone () {}
+
+    //It even has a comment
+    public function a_helper_function () {}
+
+    /**
+     * @depends depends_on_someone
```

Comment:
here, in input code, the method we depends on is still called `test_depends_on_someone`.
It shall remain `depends_on_someone` in expected code